AlienBlast

aloysboscariol opened the pull request:
# Your checklist for this pull request

## Author name

**Author:**
Aloys
<!-- A name or nickname that you want to appear as the author of the game -->

## About your game

**What is your game about?**
Killing the aliens before they invade the earth
<!-- Example: Pushing boxes to the goal. (from [Sokoban Plus](https://sprig.hackclub.com/gallery/sokoban_plus)) -->

**How do you play your game?**
Use a to move left, d to move right and w to shoot.
<!-- Example: Press WASD to move, J to restart and K to toggle trails, Get A boxes (cyan) to A goals (green), Get B boxes (magenta) to B goals (red), Get normal boxes (gray) to either goal. (from [Sokoban plus](https://sprig.hackclub.com/gallery/sokoban_plus)) -->

## Code
Check off the items that are true. <!-- Put a `x` in the  `[ ]`. Example: `[x] The game was made...` -->
- [x] The game was made using the [Sprig editor](https://sprig.hackclub.com/editor).
- [x] The game is placed in the in the [`/games` directory](https://github.com/hackclub/sprig/tree/main/games).
- [x] The code is significantly different from all other games in the [Sprig gallery](https://sprig.hackclub.com/gallery) (except for games labeled "demo").
- [x] The game runs without errors. 
- [x] The name of the file/game contains only alphanumeric characters, `-`s, or `_`s.
- [x] The game name is not the same as the others from [gallery](https://sprig.hackclub.com/gallery)
- [x] The game has metadata at the top of the file in the following format:
	>     /*
	>     @title:
	>     @author:
	>     @description:
	>     @tags: []
	>     @addedOn: 2025-MM-DD
	>     */

## Image (If an image is used)

- [x] The image is in the [`/games/img` directory](https://github.com/hackclub/sprig/tree/main/games/img).
- [x] The name of the image matches the name of your file. <!-- Example: `sokoban_plus.js` -> `sokoban_plus.png`. -->

> Thanks for your PR!

whatwareweb commented:
OK, just looked at your game again. 

First off, the game mechanics seem really nice, the physics are super clean. 

Some things to do to improve it before we merge:

- I saw the instructions you put and then removed, could you just put those in a standard `/* javascript comment block */` ?
- Improving the artwork could really give this game a level of polish. Making a background as well as some sprites for the player and enemies would be awesome. If you really want to go above and beyond to give the game some polish you might even consider adding animations if you want to do something extra

Thanks for your PR, looking good so far!

aloysboscariol commented:
I just updated the game. 
I added:
- Game menu
- 3 skins variants
- 3 maps variants
- input list down the screen
But I couldn't add animations because the game was always crashing.
I'll add:
- More instructions in the game comment
- A game thumbail (i'm working on it right now)

aloysboscariol commented:
I added the instructions in the game comment, I added the thumbnail and I fixed the bugs. Let me know if there are any errors that I need to fix :)

whatwareweb commented:
Hi, i saw the updates and that is looking good so far!

Only thing that needs to be done now before we merge is that sometimes when you shoot, it just goes right through the aliens without anything happening. Not sure what's causing that, but can you take a look? After that's fixed i'll merge

Thanks!

aloysboscariol commented:
Thanks, I will also look on my side to find the error!

aloysboscariol commented:
That's it! :)
